From f3489d2ccde3033d3cee3226c64d617c60acfc02 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 12 Dec 2013 11:34:55 +0100
Subject: [PATCH 292/311] MINOR: pattern: do not assign SMP_TYPES by default to
 patterns

This is never used since we exit on parse failure, and it's confusing.
---
 src/acl.c     | 1 -
 src/pattern.c | 1 -
 2 files changed, 2 deletions(-)

diff --git a/src/acl.c b/src/acl.c
index be20959..974e919 100644
--- a/src/acl.c
+++ b/src/acl.c
@@ -507,7 +507,6 @@ struct acl_expr *parse_acl_expr(const char **args, char **err, struct arg_list *
 		}
 		pattern->flags = patflags;
 
-		pattern->type = SMP_TYPES; /* unspecified type */
 		ret = expr->pat.parse(args, pattern, NULL, &opaque, err);
 		if (!ret)
 			goto out_free_pattern;
diff --git a/src/pattern.c b/src/pattern.c
index 084f73a..6c2a5ba 100644
--- a/src/pattern.c
+++ b/src/pattern.c
@@ -860,7 +860,6 @@ int pattern_register(struct pattern_expr *expr, char *text,
 		(*pattern)->val.tree = &expr->pattern_tree;
 	}
 
-	(*pattern)->type = SMP_TYPES; /* unspecified type by default */
 	if (!expr->parse(args, *pattern, smp, &opaque, err))
 		return -1;
 
-- 
1.8.1.5

